numba/core/types/npytypes.py

```diff
@@ -31,6 +31,11 @@ def can_convert_from(self, typingctx, other):
         if isinstance(other, Bytes):
             return Conversion.safe
 
+    def can_convert_to(self, typingctx, other):
+        if isinstance(other, CharSeq):
+            if other.count < self.count:
+                return Conversion.unsafe
+            return Conversion.safe
```

Comment:
Note for reviewer: 

- review why `UnicodeCharSeq` has a different cast logic: https://github.com/numba/numba/pull/9306/files#diff-98dd0dbec153c38b3afee6df1262d9e3fd3436dc477cc1d6b61771fb9b40bcfcL56-L59

Reply:
`UnicodeCharSeq` is lacking an if-case when `other` is also an `UnicodeCharSeq`. So, the following fails:

```python
from numba.core import types
from numba import njit
import numpy as np

src_typ = types.Array(types.UnicodeCharSeq(4), 1, "A")
dst_typ = types.Array(types.UnicodeCharSeq(5), 1, "A")
sig = types.int32(src_typ, dst_typ)

@njit(sig)
def assign_unicodecharseq_array(src, dst):
    for i in range(len(src)):
        dst[i] = src[i]
    return 0

x = np.array(["1234", "5678"], dtype="U4")
y = np.empty(2, dtype="U5")
assign_unicodecharseq_array(x, y)

# numba.core.errors.NumbaNotImplementedError: Failed in nopython mode pipeline (step: native lowering)
# Cannot cast [unichr x 4] to [unichr x 5]: %".191" = load [4 x i32], [4 x i32]* %".190"
# During: lowering "dst[i] = $22binary_subscr.4" at /Users/guilhermeleobas/git/numba/b.py (12)
```